lib/polyfill/patchedmediakeys_20140218.js

```diff
@@ -0,0 +1,800 @@
+  this.updatePromise_ = null;
+
+  /** @type {string} */
+  this.sessionId = '';
+
+  /** @type {number} */
+  this.expiration = NaN;
+
+  /** @type {!shaka.util.PublicPromise} */
+  this.closed = new shaka.util.PublicPromise();
+
+  /** @type {!MediaKeyStatusMap} */
+  this.keyStatuses =
+      new shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeyStatusMap();
+};
+goog.inherits(shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeySession,
+    shaka.util.FakeEventTarget);
+
+
+/** @override */
+shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeySession.prototype.
+    generateRequest = function(initDataType, initData) {
+  shaka.log.debug('v20140218.MediaKeySession.generateRequest');
+
+  this.generateRequestPromise_ = new shaka.util.PublicPromise();
+
+  try {
+    // This EME spec version requires a MIME content type as the 1st param
+    // to createSession, but doesn't seem to matter what the value is...
+    this.nativeMediaKeySession_ = this.nativeMediaKeys_
+        .createSession('video/mp4', initData);
+
+    // Alias
+    var v20140218 = shaka.polyfill.PatchedMediaKeys.v20140218;
+
+    // Attach session event handlers here
+    this.nativeMediaKeySession_.addEventListener('mskeymessage',
+        v20140218.onMsKeyMessage_.bind(this));
+    this.nativeMediaKeySession_.addEventListener('mskeyadded',
+        v20140218.onMsKeyAdded_.bind(this));
+    this.nativeMediaKeySession_.addEventListener('mskeyerror',
+        v20140218.onMsKeyError_.bind(this));
+
+    this.updateKeyStatus_('status-pending');
+  }
+  catch (exception) {
+    this.generateRequestPromise_.reject(exception);
+  }
+
+  return this.generateRequestPromise_;
+};
+
+
+/** @override */
+shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeySession.prototype.
+    load = function() {
+  shaka.log.debug('v20140218.MediaKeySession.load');
+
+  return Promise.reject(new Error('MediaKeySession.load is only ' +
+      'applicable for persistent licenses, which are not supported ' +
+      'on this platform'));
+};
+
+
+/** @override */
+shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeySession.prototype.
+    update = function(response) {
+  shaka.log.debug('v20140218.MediaKeySession.update');
+
+  this.updatePromise_ = new shaka.util.PublicPromise();
+
+  try {
+    // Pass through to the native session
+    this.nativeMediaKeySession_.update(response);
+  }
+  catch (exception) {
+    this.updatePromise_.reject(exception);
+  }
+
+  return this.updatePromise_;
+};
+
+
+/** @override */
+shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeySession.prototype.
+    close = function() {
+  shaka.log.debug('v20140218.MediaKeySession.close');
+
+  try {
+    // Pass through to the native session
+    // NOTE: IE seems to have spec discrepancy here - v2010218 should have
+    // MediaKeySession.release, but uses "close". The next version of the
+    // spec is the initial Promise based one, so it's not the target spec
+    // either. Am supporting both just in case..
```

Comment:
Since the installation of this polyfill is predicated on MSMediaKeys, I think it is safe to just support what IE11 supports and make a note of the fact that this bit is off-spec for the version MS implemented.